### PR TITLE
Implement the navigation and reduced navigation 

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- Implement the [navigation](https://vanillaframework.io/docs/examples/patterns/navigation/search-dark?theme=light) and [reduced navigation](https://vanillaframework.io/docs/examples/patterns/navigation/reduced?theme=light) HTML patterns
- Add Python logic to handle secondary navigation
- Add JS to handle the states of search and secondary navigation (dropdowns, mobile menu don't work). Most is boiler plate JS copied from Vanilla, u.com and the other c.com navigation

## QA

- Go https://canonical-com-1290.demos.haus/ to and see the implementation of the regular navigation matches the [vanilla navigation](https://vanillaframework.io/docs/examples/patterns/navigation/search-dark?theme=light)
- Go to https://canonical-com-1290.demos.haus/data and see the implementation of the reduced navigation matches the [reduced navigation](https://vanillaframework.io/docs/examples/patterns/navigation/reduced?theme=light)
- Check the search opens and closes as expected on the normal navigation, reduced navigation and mobile (for both)
- Check that on mobile, the secondary navigation dropdown functions properly

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-12668

## Screenshots
Regular navigation:
image
Reduced navigation:
image
Regular navigation (search open):
image
Reduced navigation (search open):
image

### CODE REVIEW

**Comment by akbarkz** [comment]
@petesfrench Two findings with the Search functionality:
1. The background of the search field is black, while as on ubuntu.com it's white
2. The focus is not automatically put to the search field when you open it

File: static/js/navigation.js [hunk 1]
```diff
@@ -1,269 +1,375 @@
...
-          }
-        }
-      } else {
-        dropdown.classList.remove("is-active");
-        overlay.classList.remove("is-applied");
-      }
-    });
-  });
-});
+// // Careers meganav variables
```
Comment (akbarkz) [inline static/js/navigation.js:107]: Can you add comments on why this piece of code is commented out? Or just remove it?

Comment (petesfrench) [reply static/js/navigation.js:107]: I was keeping it for practicality as I want to use some of it. But I suppose I can just use whats on `main' instead. Removed.

**Comment by petesfrench** [comment]
@akbarkz

1. This is how appears on Vanilla with the `is-dark` them applied to the page, same as c.com. We should be aligning with Vanilla as much as possible, not u.com.
2. Fixed.

Can you take another look please?

**Comment by akbarkz** [comment]
@petesfrench After typing something into the search field and clicking on "X" icon to erase it, it doesn't work.

**Comment by petesfrench** [comment]
@akbarkz I think that is getting into the [search implementation](https://warthogs.atlassian.net/browse/WD-12996). My thought process behind this is that 'clear search' is something I would implement with the 'search initialization', along with the event listener for the search button, keyboard events, search functionality etc.

**Review by akbarkz — APPROVED** [review]

**Pull request merged**